task/tekton-operator-install/0.1/tekton-operator-install.yaml

```diff
@@ -0,0 +1,187 @@
+          echo "failed to install $1 $2"
+          exit 1
+        }
+
+        main() {
+          local maxloop=30 # 5 minutes max
+          local count=0
+          local trigger_success=true
+          local dashboard_success=true
+          local extension_success=true
+
+          [[ $(params.INSTALL_TRIGGERS) == "false" ]] && \
+          [[ $(params.INSTALL_DASHBOARD) == "false" ]] && \
+          [[ $(params.INSTALL_EXTENSION_WEBHOOKS) == "false" ]] && \
+          echo "No addons to be installed" && exit 0
+
+          [[ $(params.INSTALL_TRIGGERS) == "true" ]] && \
+          createCustomResource "trigger" && trigger_success=false
+          [[ $(params.INSTALL_DASHBOARD) == "true" ]] && \
+          kubectl create clusterrolebinding tekton-operator-cluster-admin \
+            --clusterrole cluster-admin --serviceaccount tekton-operator:tekton-operator && \
+          createCustomResource "dashboard" && dashboard_success=false
+          [[ $(params.INSTALL_EXTENSION_WEBHOOKS) == "true" ]] && \
+          createCustomResource "extensionwebhooks" && extension_success=false
+
+          while true; do
+
+            # hard timeout condition
+            if [[ ${count} == ${maxloop} ]]; then
+              # check for trigger failure after timeout
+              [[ "$trigger_success" == "false" ]] && show_failure "tektonaddon" "trigger"
+              # check for dashboard failure after timeout
+              [[ "$dashboard_success" == "false" ]] && show_failure "tektonaddon" "dashboard"
+              # check for extension webhooks failure in case of timeout
+              [[ "$extension_success" == "false" ]] && show_failure "tektonaddon" "extensionwebhooks"
+            fi
+
+            #if Trigger needs to be installed then check the status
+            [[ $(params.INSTALL_TRIGGERS) == "true" ]] && [[ $trigger_success == "false" ]] && \
+            [[ $(checkCodeStatus "tektonaddon" "trigger") == "installed" ]] && \
+            checkVersionStatus "tektonaddon" "trigger" && trigger_success=true
+
+            #if dashboard needs to be installed then check the status
+            [[ $(params.INSTALL_DASHBOARD) == "true" ]] && [[ dashboard_success == "false" ]] && \
+            [[ $(checkCodeStatus "tektonaddon" "dashboard") == "installed" ]] && \
+            checkVersionStatus "tektonaddon" "dashboard" && dashboard_success=true
+
+            #if extension webhooks needs to be installed then check the status
+            [[ $(params.INSTALL_EXTENSION_WEBHOOKS) == "true" ]] && [[ $extension_success == "false" ]] && \
+            [[ $(checkCodeStatus "tektonaddon" "extensionwebhooks") == "installed" ]] && \
+            checkVersionStatus "tektonaddon" "extensionwebhooks" && extension_success=true
+
+            # break if required Addons are successfully installed
+            [[ "$trigger_success" == "true" ]] && [[ "$dashboard_success" == "true" ]] && \
+            [[ "$extension_success" == "true" ]] && break
```

Comment:
shall we break this into 3 different checks, 
if the TaskRun param is true, then we can check `<item>_success` 
this could be handled as a function as well.